### ISSUE TITLE
false positive, ComponentChain, not a structure (check_syntax) [E]

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#1488]
```abap
CLASS lcl_clas DEFINITION.
  PUBLIC SECTION.
    DATA:
      BEGIN OF settings READ-ONLY,
        field TYPE abap_bool,
      END OF settings.
ENDCLASS.
CLASS lcl_clas IMPLEMENTATION.
ENDCLASS.

DATA(bar) = NEW lcl_clas( )->settings.
```

![image](https://user-images.githubusercontent.com/5888506/97149991-c641b080-176d-11eb-9130-8a900b63b122.png)
